### PR TITLE
GAWB-1260 Backports from Orchestration to rawlsModel

### PR DESCRIPTION
**Pull request by helgridly**
This gives us a build of rawlsModel that we can use in Orchestration, a prerequisite for GAWB-1260.

- [x] **Submitter**: Include the JIRA issue number in the PR description
- [x] **Submitter**: Check that the **Product Owner** has signed off on any user-facing changes
- [x] **Submitter**: Make sure Swagger is updated if API changes
  - [x] **...and Orchestration's Swagger too!**
- [x] **Submitter**: If updating admin endpoints, also update [firecloud-admin-cli](https://github.com/broadinstitute/firecloud-admin-cli)
- [x] **Submitter**: Check documentation and code comments. Add explanatory PR comments if helpful.
- [x] **Submitter**: JIRA ticket checks:
  * Acceptance criteria exists and is met
  * Note any changes to implementation from the description
  * Add notes on what you've tested
- [x] **Submitter**: Update RC_XXX release ticket with any config or environment changes necessary
- [x] **Submitter**: Database checks:
  * If PR includes new or changed db queries, include the explain plans in the description
  * Make sure liquibase is updated if appropriate
  * If doing a migration, take a backup of the
  [dev](https://console.cloud.google.com/sql/instances/terraform-qfarbdq3lrexxck5htofjs5z6m/backups?project=broad-dsde-dev&organizationId=548622027621)
  and
  [alpha](https://console.cloud.google.com/sql/instances/terraform-r4caezzc35c4tb7pgdhwkmme4y/backups?project=broad-dsde-alpha&organizationId=548622027621)
  DBs in Google Cloud Console
- [x] **Submitter**: Update FISMA documentation if changes to:
  * Authentication
  * Authorization
  * Encryption
  * Audit trails
- [x] **Submitter**: If you're adding new libraries, sign us up to security updates for them
- [x] Tell your tech lead (TL) that the PR exists if they want to look at it
- [x] Anoint a lead reviewer (LR). **Assign PR to LR**
* Review cycle:
  * LR reviews
  * Rest of team may comment on PR at will
  * **LR assigns to submitter** for feedback fixes
  * Submitter rebases to develop again if necessary
  * Submitter makes further commits. DO NOT SQUASH
  * Submitter updates documentation as needed
  * Submitter **reassigns to LR** for further feedback
- [x] **TL** sign off
- [x] **LR** sign off
- [x] **Assign to submitter** to finalize
- [x] **Submitter**: Verify all tests go green, including CI tests
- [x] **Submitter**: Squash commits and merge to develop
- [x] **Submitter**: Delete branch after merge
- [x] **Submitter**: **Test this change works on dev environment after deployment**. YOU own getting it fixed if dev isn't working for ANY reason!
- [x] **Submitter**: Verify swagger UI on dev environment still works after deployment
- [x] **Submitter**: Inform other teams of any API changes via hipchat and/or email
- [x] **Submitter**: Mark JIRA issue as resolved once this checklist is completed

### CODE REVIEW

File: core/src/main/scala/org/broadinstitute/dsde/rawls/webservice/MethodConfigApiService.scala [hunk 1]
```diff
@@ -64,8 +63,11 @@ trait MethodConfigApiService extends HttpService with PerRequestCreator with Use
     path("workspaces" / Segment / Segment / "methodconfigs" / Segment / Segment) { (workspaceNamespace, workspaceName, methodConfigurationNamespace, methodConfigName) =>
       put {
         entity(as[MethodConfiguration]) { newMethodConfiguration =>
-          requestContext => perRequest(requestContext, WorkspaceService.props(workspaceServiceConstructor, userInfo),
-            WorkspaceService.UpdateMethodConfiguration(WorkspaceName(workspaceNamespace, workspaceName), newMethodConfiguration.copy(namespace = methodConfigurationNamespace, name = methodConfigName)))
+          requestContext => {
+            val copyMC = newMethodConfiguration.copy(namespace = methodConfigurationNamespace, name = methodConfigName)
+            perRequest(requestContext, WorkspaceService.props(workspaceServiceConstructor, userInfo),
+              WorkspaceService.UpdateMethodConfiguration(WorkspaceName(workspaceNamespace, workspaceName), newMethodConfiguration.copy(namespace = methodConfigurationNamespace, name = methodConfigName)))
```
Comment (breilly2) [inline core/src/main/scala/org/broadinstitute/dsde/rawls/webservice/MethodConfigApiService.scala]: Did you mean for `copyMC` to be used here?

File: core/src/test/scala/org/broadinstitute/dsde/rawls/mock/RemoteServicesMockServer.scala [hunk 2]
```diff
@@ -551,5 +551,25 @@ class RemoteServicesMockServer(port:Int) extends RawlsTestUtils {
       )
   }
 
+  mockServer.when(
```
Comment (breilly2) [inline core/src/test/scala/org/broadinstitute/dsde/rawls/mock/RemoteServicesMockServer.scala:554]: I don't like how far away this is from the test that cares about this response, but I think I mentioned that in the previous PR and you said you'd rather defer that refactoring.

File: model/src/main/scala/org/broadinstitute/dsde/rawls/model/WorkspaceModel.scala [hunk 3]
```diff
@@ -436,7 +449,9 @@ object WorkspaceJsonSupport extends JsonSupport {
     }
   }
 
-  implicit val ErrorReportFormat: RootJsonFormat[ErrorReport] = rootFormat(lazyFormat(jsonFormat(ErrorReport,"source","message","statusCode","causes","stackTrace","exceptionClass")))
+  implicit val ErrorReportFormat: RootJsonFormat[ErrorReport] = rootFormat(lazyFormat(jsonFormat(ErrorReport.apply,"source","message","statusCode","causes","stackTrace","exceptionClass")))
```
Comment (breilly2) [inline model/src/main/scala/org/broadinstitute/dsde/rawls/model/WorkspaceModel.scala:452]: I'm guessing that this at least part of the magic that allowed you to use `ErrorReport` the way you wanted to?

File: core/src/main/scala/org/broadinstitute/dsde/rawls/webservice/MethodConfigApiService.scala (hunk 1 above)
Comment (helgridly) [reply core/src/main/scala/org/broadinstitute/dsde/rawls/webservice/MethodConfigApiService.scala]: No, that looks like dead code.

File: core/src/test/scala/org/broadinstitute/dsde/rawls/mock/RemoteServicesMockServer.scala (hunk 2 above)
Comment (helgridly) [reply core/src/test/scala/org/broadinstitute/dsde/rawls/mock/RemoteServicesMockServer.scala:554]: If I refactor ever piece of bad code I see, I'll never end up checking this in :)

Comment (breilly2) [reply core/src/test/scala/org/broadinstitute/dsde/rawls/mock/RemoteServicesMockServer.scala:554]: I get it. Incremental improvement. :)

File: core/src/main/scala/org/broadinstitute/dsde/rawls/webservice/MethodConfigApiService.scala (hunk 1 above)
Comment (breilly2) [reply core/src/main/scala/org/broadinstitute/dsde/rawls/webservice/MethodConfigApiService.scala]: Cool, I'll expect line 67 go away before merging.

File: model/src/main/scala/org/broadinstitute/dsde/rawls/model/WorkspaceModel.scala (hunk 3 above)
Comment (helgridly) [reply model/src/main/scala/org/broadinstitute/dsde/rawls/model/WorkspaceModel.scala:452]: Yup. The other part of the magic is package.scala above.

**Review by breilly2 — APPROVED** [review]
👍

**Pull request merged**